frontend/src/components/vuetify/NavigationDrawer.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <v-navigation-drawer
+    v-model="drawer"
+    :location="computedLocation"
+    color="grey-lighten-4"
+    temporary
+    width="400"
+    class="custom-drawer"
+    mobile
+    mobile-breakpoint="md"
+  >
+    <slot />
+  </v-navigation-drawer>
+</template>
+
+<script lang="ts" setup>
+import { computed, ref, watch, PropType } from 'vue'
+import { useDisplay } from 'vuetify'
+
+type ValidLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'
+
+const props = defineProps({
+  modelValue: {
+    type: Boolean,
+    required: true,
+  },
+  location: {
+    type: String as PropType<NavigationDrawerLocation>,
+    required: false,
+    default: 'right',
+    validator: (value: string) =>
+      ['right', 'bottom', 'left', 'end', 'top', 'start'].includes(value as ValidLocation),
+  },
+})
+
+const emits = defineEmits(['update:modelValue'])
+const display = useDisplay()
+const drawer = ref(props.modelValue)
+
+const computedLocation = computed(() => {
+  return display.mobile.value
+    ? 'bottom'
+    : ['right', 'bottom', 'left', 'end', 'top', 'start'].includes(props.location)
+      ? props.location
+      : 'right'
+})
```

Comment:
```suggestion
const computedLocation = computed(() => {
  return display.mobile.value
    ? 'bottom'
    : props.location
})
```

I don't think the additional check is necessary.

Reply:
changed